### PR TITLE
Migrate Singapore holidays to holiday groups

### PR DESCRIPTION
**Pull request by KJhellico**
## Proposed change

Migrate Singapore holidays to `holiday_groups` standard.

"Vesak Day" calculated as 15th day of 4th month by Chinese calendar, "Deepavali" - as (1st day of 10th month) - 2 days. It's exactly equal to `_ChineseLuniSolar::vesak_date()` and `_ChineseLuniSolar::s_diwali_date()`.

## Type of change

- [ ] New country/market holidays support (thank you!)
- [ ] Supported country/market holidays update (calendar discrepancy fix, localization)
- [x] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency upgrade (version update)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] This PR is filed against `beta` branch of the repository
- [x] This PR doesn't contain any merge conflicts and has clean commit history
- [x] The code style looks good: `make pre-commit`
- [x] All tests pass locally: `make test`, `make tox` (we strongly encourage adding tests to your code)
- [x] The related [documentation][docs] has been added/updated (check off the box for free if no updates is required)

[contributing-guidelines]: https://github.com/dr-prodigy/python-holidays/blob/beta/CONTRIBUTING.rst
[docs]: https://github.com/dr-prodigy/python-holidays/tree/beta/docs/source

### CODE REVIEW

File: holidays/countries/singapore.py [hunk 1]
```diff
@@ -276,24 +284,42 @@ def _add_with_observed(
...
+        # the day next following not being itself a public holiday
+        # is declared a public holiday in Singapore."
+        if self.observed and year >= 1998:
+            for dt in sorted(observed_dates):
+                if not self._is_sunday(dt):
+                    continue
+                obs_date = dt + td(days=+1)
+                if obs_date in observed_dates:
+                    obs_date += td(days=+1)
+                self._add_holiday("%s (Observed)" % self[dt], obs_date)
```
Comment (arkid15r) [inline holidays/countries/singapore.py:318]:
```suggestion
                self._add_holiday(
                    "%s (Observed)" % self[dt],
                    dt
                    + td(days=2 if dt + td(days=+1) in observed_dates else 1),
                )

```

File: holidays/countries/singapore.py [hunk 2]
```diff
@@ -276,24 +284,42 @@ def _add_with_observed(
...
 
         # special case (observed from previuos year)
         if self.observed and year == 2007:
-            self[date(2007, JAN, 2)] = "Hari Raya Haji (Observed)"
+            self._add_holiday("Hari Raya Haji (Observed)", JAN, 2)
```
Comment (arkid15r) [inline holidays/countries/singapore.py:322]: The if above can be simplified and moved under previous one.

// also fix the comment typo please

**Review by arkid15r — COMMENTED** [review]
🇸🇬 LGTM

File: holidays/countries/singapore.py (hunk 1 above)
Comment (KJhellico) [reply holidays/countries/singapore.py]: Done.

File: holidays/countries/singapore.py (hunk 2 above)
Comment (KJhellico) [reply holidays/countries/singapore.py:325]: Done.

**Pull request merged**